daemon/ipam.go

```diff
@@ -194,7 +194,7 @@ func (d *Daemon) allocateDatapathIPs(family datapath.NodeAddressingFamily) (rout
 
 func (d *Daemon) allocateHealthIPs() error {
 	bootstrapStats.healthCheck.Start()
-	if option.Config.EnableHealthChecking {
+	if option.Config.EnableEndpointHealthChecking {
```

Comment:
this should then be `option.Config.EnableHealthChecking && option.Config.EnabledEndpointHealthChecking` because one disables the other.

Reply:
Why? The health IP is only needed if we perform endpoint health checking. When health checking is enabled and endpoint health checking is disabled, no health IP is required. All health checking will be done across node IPs.

Reply:
>The health IP is only needed if we perform endpoint health checking.

Correct

>When health checking is enabled and endpoint health checking is disabled, no health IP is required.

Also correct.

Why should any kind of health checking, including endpoint health checking, be enabled if `EnableHealthChecking` is set to false?

Reply:
Sorry, now I understand what you mean. You are absolutely right.